### DISCUSSION TITLE
Should Leon perform HTTP requests?

### DISCUSSION THREAD

**Discussion by svenjacobs** [leon-cleaning-services/leon#80] [Ideas]
Recently there have been two features requests that require Leon to perform HTTP requests, namely unwrapping [Amazon](https://github.com/svenjacobs/leon/issues/75) as well as [TikTok](https://github.com/svenjacobs/leon/issues/77) short URLs. The URLs, in contrast to for example [YouTube](https://github.com/svenjacobs/leon/issues/56) short URLs, cannot be unwrapped locally. Leon has to perform a HTTP request to receive the actual URL.

Currently I'm not sure if this is something that Leon should be doing. As of now, Leon can perform all unwrapping and cleaning locally. If Leon should be able to do this, the following changes have to be made:

- Ask the user before performing a network request
- Show a progress dialog / screen after a link was sent to Leon and before the result is displayed, because HTTP requests might take a few seconds
- Handle possible errors of HTTP requests
- Prioritise sanitizer rules because unwrapping should happen before other rules are applied

Asking some power users of Leon, what do you think about this? @TPS @slootsky @Enkidu70

**Comment by TPS** [comment]
@svenjacobs Imho, if you spend the time/energy on this (#75, #77, &c), then I'd like to see some substantial safeguards.

0. Maybe split this off into another app entirely?   That way, this 1 can stay focused on its strengths?  Or just update/recommend, e.g., https://f-droid.org/app/com.github.nicolassmith.urlevaluator ← https://github.com/nicolassmith/urlevaluator.

Else:

1. The settings page is redesigned such that such processing is **clearly** labeled w/ something like "All the below cleaners require online/HTTP processing & therefore **_will_** allow some level of tracking."

2. All of these settings are off/inaccessible by default.

3. Every time that 1 attempts to use any of these, a popup (w/ message from safeguard 1) & option to abort is required

4. These must be anonymized as much as possible during processing to minimize whatever tracking.  ←  The way these are designed, I've no idea how possible this is.

**Reply by svenjacobs** [reply]
I'm absolutely with you @TPS and I don't really like the idea either. I think Leon should stay close to its root so I'm going to close these issues.
